test/models/ocelot/stanzas/ChoiceStanzaSpec.scala

```diff
@@ -187,11 +187,189 @@ class ChoiceStanzaSpec extends BaseSpec {
+      val choice = Choice(stanza)
+      val lc = LabelCache()
+      val expectedResult = (Seq("41"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when no tests succeed" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("4", LessThanOrEquals, "3"),
+                                                        ChoiceStanzaTest("3", MoreThan, "4")), false)
+      val choice = Choice(stanza)
+      val lc = LabelCache()
+      val expectedResult = (Seq("50"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when one of the tests succeed referencing labels" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("[label:X]", LessThanOrEquals, "[label:Y]"),
+                                                        ChoiceStanzaTest("3", LessThanOrEquals, "4"),
+                                                        ChoiceStanzaTest("3", NotEquals, "4")), false)
+      val choice = Choice(stanza)
+      val labels = Map("X"->Label("X", Some("33.5")), "Y"->Label("Y", Some("44")))
+      val lc = LabelCache(labels)
+      val expectedResult = (Seq("40"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+    "Evaluate to correct result when no tests succeed referencing labels" in {
+      val stanza: ChoiceStanza = ChoiceStanza(next, Seq(ChoiceStanzaTest("[label:X]", LessThanOrEquals, "[label:Y]"),
+                                                        ChoiceStanzaTest("3", Equals, "4"),
+                                                        ChoiceStanzaTest("3", MoreThanOrEquals, "4")),
+                                                        false)
+      val choice = Choice(stanza)
+      val labels = Map("X"->Label("X", Some("33.5")), "Y"->Label("Y", Some("4")))
+      val lc = LabelCache(labels)
+      val expectedResult = (Seq("50"), lc)
+      choice.eval(lc) shouldBe expectedResult
+    }
+
+  }
+
   "ChoiceTest" must {
+    "provide support to EqualsTest" in {
+      EqualsTest("5", "5").eval(LabelCache()) shouldBe true
+
+      EqualsTest("4", "5").eval(LabelCache()) shouldBe false
+
+      EqualsTest("hello", "hello").eval(LabelCache()) shouldBe true
+
+      EqualsTest("4", "hello").eval(LabelCache()) shouldBe false
+    }
+
+    "provide support to NotEqualsTest" in {
+      NotEqualsTest("5", "5").eval(LabelCache()) shouldBe false
+
+      NotEqualsTest("4", "5").eval(LabelCache()) shouldBe true
+
+      NotEqualsTest("hello", "hello").eval(LabelCache()) shouldBe false
+
+      NotEqualsTest("4", "hello").eval(LabelCache()) shouldBe true
+    }
+
+    "provide support to MoreThanTest" in {
+      MoreThanTest("5", "5").eval(LabelCache()) shouldBe false
+
+      MoreThanTest("4", "5").eval(LabelCache()) shouldBe false
+
+      MoreThanTest("4", "3").eval(LabelCache()) shouldBe true
+
+      MoreThanTest("hello", "hello").eval(LabelCache()) shouldBe false
```

Comment:
Are tests such as MoreThan appropriate for strings such as "hello". Should we be raising an error? What does Ocelot do?

Reply:
It does the char/string equivalent as here i.e. a < b. I havent seen Ocelot generate any runtime errors, it works with what its given or does nothing as in the case of invalid or no input. If the guidance allows input as valid, dont think these tests should generate any runtime errors

Reply:
As we discussed Scala probably follows the same approach as Java and Javascript when applying relational operators to strings.